mars/core/operand/base.py

```diff
@@ -153,6 +157,30 @@ def __init__(self: OperandType, *args, **kwargs):
         self._extract_scheduling_hint(kwargs)
         super().__init__(*args, **kwargs)
 
+    def _update_key(self):
+        super()._update_key()
+        if not hasattr(self, "logic_key") or not self.logic_key:
+            self._obj_set("logic_key", self._get_logic_key())
```

Comment:
Do we need to generate such a `logic_key` every time creating an operand? How about generating it when a subtask is created?

Reply:
`logic_key` generation can be made lazy. How about:
```
    @functools.lru_cache
    def get_logic_key(self):
        """The subclass may need to override this method to ensure unique and deterministic."""
        fields = self._get_logic_key_token_values()
        try:
            return tokenize(*fields)
        except Exception as e:  # pragma: no cover
            raise Exception(
                f"Cannot generate logic key for operator {self} with fields {fields}"
            ) from e
```